### PR TITLE
[DiffDrive] Fix warnings

### PR DESCRIPTION
**Pull request by artivis**
- Fix a signed/unsigned comparison
- CMake include catkin includes as system to avoid warnings spamming from other pkgs

### CODE REVIEW

**Comment by artivis** [comment]
@bmagyar @efernandez Hola.

File: diff_drive_controller/CMakeLists.txt [hunk 1]
```diff
@@ -15,7 +15,11 @@ catkin_package(
 )
 
 include_directories(
-  include ${catkin_INCLUDE_DIRS}
+  include
+)
+
+include_directories(SYSTEM
```
Comment (bmagyar) [inline diff_drive_controller/CMakeLists.txt:21]: Could you elaborate on this please? What kind of warnings?

Comment (artivis) [reply diff_drive_controller/CMakeLists.txt:21]: Not diff-drive-related compilation warnings e.g. 
```terminal
/opt/ros/kinetic/include/hardware_interface/internal/interface_manager.h:69:85: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   static const void callConcatManagers(typename std::vector<T*>& managers, T* result)
                                                                                     ^
In file included from /opt/ros/kinetic/include/controller_interface/controller_base.h:37:0,
                 from /opt/ros/kinetic/include/controller_manager/controller_spec.h:41,
                 from /opt/ros/kinetic/include/controller_manager/controller_manager.h:36,
                 from /home/jeremiederay/workspaces/ros_controllers/src/ros_controllers/diff_drive_controller/test/diffbot.cpp:34:
/opt/ros/kinetic/include/hardware_interface/robot_hw.h:80:38: warning: unused parameter 'root_nh' [-Wunused-parameter]
   virtual bool init(ros::NodeHandle& root_nh, ros::NodeHandle &robot_hw_nh) {return true;}
                                      ^
/opt/ros/kinetic/include/hardware_interface/robot_hw.h:80:64: warning: unused parameter 'robot_hw_nh' [-Wunused-parameter]
   virtual bool init(ros::NodeHandle& root_nh, ros::NodeHandle &robot_hw_nh) {return true;}
```

Comment (bmagyar) [reply diff_drive_controller/CMakeLists.txt:21]: :+1:

**Review by bmagyar — APPROVED** [review]

File: diff_drive_controller/CMakeLists.txt (hunk 1 above)
Comment (efernandez) [reply diff_drive_controller/CMakeLists.txt:21]: Yes, I've also started to do this. :+1:

File: diff_drive_controller/CMakeLists.txt [hunk 2]
```diff
@@ -15,7 +15,11 @@ catkin_package(
...
+ ${catkin_INCLUDE_DIRS}
```
Comment (efernandez) [inline diff_drive_controller/CMakeLists.txt]: pedantic: This should be indented by two spaces.

**Review by efernandez — CHANGES_REQUESTED** [review]

File: diff_drive_controller/CMakeLists.txt (hunk 2 above)
Comment (artivis) [reply diff_drive_controller/CMakeLists.txt]: fixed and force pushed.

**Review by efernandez — APPROVED** [review]
Thanks

**Comment by artivis** [comment]
#292 again ??

I'm having a look at it btw.

**Comment by bmagyar** [comment]
Thank you for your cooperation citizen.

**Pull request merged**